src/state/opamUpdate.ml

```diff
@@ -86,7 +86,7 @@ let repository rt repo =
           reason;
         job { r with repo_url = new_url } (n-1)
   in
-  job repo max_loop @@+ fun repo ->
+  job repo max_loop @@+ fun has_changes ->
```

Comment:
```suggestion
  job repo max_loop @@+ fun repo, has_changes ->
```
`repo` is not return by the `job` function anymore. I belive this would break repositories with redirections from a glance at the code. Could you either return `repo` as before in a tuple or inside the new `[no_changes | changes]` state (whichever you think is better)

Reply:
you are right of course! I had missed the fact that the repo is returned by `job` to account for redirections. should be fixed now.